test/fail_compilation/issue20123.d

```diff
@@ -0,0 +1,44 @@
+/* TEST_OUTPUT:
+---
+fail_compilation/issue20123.d(39): Error: can only `@disable` opUnaryRight
+fail_compilation/issue20123.d(39): Error: can only `@disable` opUnaryRight
```

Comment:
`A` doesn't have disabled methods?

Reply:
as in the only thing you are allowed to do with `opUnaryRight` is to `@disable` it. Perhaps that should be worded better.

Reply:
> as in the only thing you are allowed to do with `opUnaryRight` is to `@disable` it.

For the moment. Give it some days and people in the forums will demand it to be usable properly.